### PR TITLE
feat: ECR repos check script ignore repos whose names contain "test".

### PR DESCRIPTION
**Pull request by artsyjian**
The type of this PR is: **Feat**

This PR solves [PHIRE-210]

### Description

- Ignore repos whose names contain "test".

Un-related re-factoring:
- Delete dummy `src/ecr_check_repos_for_terraform/test`.
- Delete `src/ecr_check_repos_for_terraform/ecr_check_repos_for_terraform/config.py`. Move setup code to main script. `config.py` was introduced to make it easier for modules to access app-wide configs. This was useful because originally scripts had many functions. So for example, `src/scriptX` would have multiple files each with multiple functions each having to access an app-wide config. Without `config.py`, the functions had to pass the configs around. But this is no longer the case. Currently most functions have been migrated to `src/lib` directory, so most scripts have only a few functions, making the advantage of `config.py` moot.

[PHIRE-210]: https://artsyproduct.atlassian.net/browse/PHIRE-210?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by joeyAghion** [comment]
Are there many test repos? Maybe it's fine to alert about them, because some action _is_ required to clean them up eventually.

**Comment by artsyjian** [comment]
> Are there many test repos? Maybe it's fine to alert about them, because some action _is_ required to clean them up eventually.

[I see only 3](https://us-east-1.console.aws.amazon.com/ecr/repositories?region=us-east-1). Yeah, a clean up job might be good, as a separate job.

File: src/lib/ecr_repos.py [hunk 1]
```diff
@@ -26,11 +26,23 @@ def all_repos(self):
       [repo.name for repo in self._repos]
     )
 
-  def get_repo(self, repo_name):
-    ''' return ECRRepo object whose name matches repo_name '''
+  def repos_with_name(self, str1, exact_match=False):
```
Comment (amonkhouse) [inline src/lib/ecr_repos.py:29]: `str1` variable would benefit from a more descriptive name

File: src/lib/test/test_ecr_repos.py [hunk 2]
```diff
@@ -28,17 +28,17 @@ def describe_instantiation():
...
-      assert len(ecr_repos._repos) == 2
+      assert len(ecr_repos._repos) == 3
       assert isinstance(repo1, ECRRepo)
   def describe_all_repos():
     def it_returns_all_repos(mock_ecr_interface):
       ecr_repos = ECRRepos(mock_ecr_interface)
-      assert set(ecr_repos.all_repos()) == set(['foo2', 'foo1'])
-  def describe_get_repo():
+      assert set(ecr_repos.all_repos()) == set(['foo1', 'foo2', 'bar1'])
+  def describe_repos_with_name():
```
Comment (amonkhouse) [inline src/lib/test/test_ecr_repos.py:37]: the default behaviour of the test has changed quite a bit (exact matches vs contains) so would benefit from more thorough testing

**Review by amonkhouse — CHANGES_REQUESTED** [review]
I think this is a great update! I left some comments about readability and logic.

File: src/lib/ecr_repos.py [hunk 3]
```diff
@@ -26,11 +26,23 @@ def all_repos(self):
...
-      if repo.name == repo_name:
-        return repo
+      if exact_match:
+        if repo.name == str1:
+          repos += [repo.name]
+          break
+      else:
+        if str1 in repo.name:
+          repos += [repo.name]
+    return repos
```
[the remark this answers is not in the document]
Comment (artsyjian) [reply src/lib/ecr_repos.py]: i like it. thanks!

File: src/lib/ecr_repos.py (hunk 1 above)
Comment (artsyjian) [reply src/lib/ecr_repos.py:29]: sure, any suggestions?

File: src/lib/test/test_ecr_repos.py (hunk 2 above)
Comment (artsyjian) [reply src/lib/test/test_ecr_repos.py:37]: right on!

File: src/lib/ecr_repos.py [hunk 4]
```diff
@@ -26,11 +26,23 @@ def all_repos(self):
...
+    always return a list
+    '''
+    repos = []
     for repo in self._repos:
-      if repo.name == repo_name:
-        return repo
+      if exact_match:
+        if repo.name == str1:
+          repos += [repo.name]
+          break
```
[the remark this answers is not in the document]
Comment (artsyjian) [reply src/lib/ecr_repos.py]: yeah, my insistence to return a list makes this confusing. the expectation is that only one repo will match, as this is an exact match, and no two repos can have the same name, therefore it's safe to break out of the loop. anyways, it's list comprehension now.

File: src/lib/test/test_ecr_repos.py [hunk 5]
```diff
@@ -28,17 +28,31 @@ def describe_instantiation():
...
+    def exact_match():
+      def it_matches(mock_ecr_interface):
+        ecr_repos = ECRRepos(mock_ecr_interface)
+        repos = ecr_repos.repos_with_name('foo1')
+        assert repos == ['foo1']
+      def it_does_not_match(mock_ecr_interface):
+        ecr_repos = ECRRepos(mock_ecr_interface)
+        repos = ecr_repos.repos_with_name('foo')
+        assert repos == []
   def describe_repos_wit_tag():
```
Comment (amonkhouse) [inline src/lib/test/test_ecr_repos.py:56]: lol I love this typo, very new york

File: src/lib/ecr_repos.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (amonkhouse) [reply src/lib/ecr_repos.py]: aaaah that makes so much sense

File: src/lib/ecr_repos.py (hunk 1 above)
Comment (amonkhouse) [reply src/lib/ecr_repos.py:29]: maybe `string_to_match`?

Comment (artsyjian) [reply src/lib/ecr_repos.py:29]: works for me.

**Pull request merged**